Makefile

```diff
@@ -153,6 +153,11 @@ pkg/apis/%/.client-gen.stamp: .k0sbuild.docker-image.k0s hack/tools/boilerplate.
 	  --output-package=github.com/k0sproject/k0s/pkg/apis/$(gen_output_dir)/
 	touch -- '$@'
 
+codegen_targets += hack/.copyright.stamp
+hack/.copyright.stamp:
+	hack/copyright.sh
+	touch -- '$@'
```

Comment:
I wonder if we should make this a PHONY target à la `lint-copyright`. It has no dependencies declared, so it will almost never be invoked again. We can invoke it explicitly in the lint CI job and maybe mention it in the contribution section in the docs.

Reply:
> I wonder if we should make this a PHONY target à la lint-copyright

Actually I considered this, but I don't think we can do this because it's one of the codegen_targets. The executions of controller-gen create the auto generated files with the copyright in the header using the static file `hack/tools/boilerplate.go.txt`. This header has a year which this script must fix, so this is called in the target `codegen`, if this target is removed, then make codegen won't set the correct copyright.

Other option would be calling the script in both codegen and lint, but the script about 12-13 seconds in my machine so that seems wasteful.

I considered making it depend on `$(shell find cmd hack internal inttest pkg static 1 -type f -name \*.go)` but this would mean that every time we make a little change and run make k0s you're rerunning the whole code which seems fairly unreasonable because in my machine it takes around 12 seconds to complete.

What we should probably do is make it depend on `pkg/apis/*/.controller-gen.stamp`.

What do you think?

> We can invoke it explicitly in the lint CI job
I don't think this is necessary because being one of the `codegen_targets` it's already executed in the CI. We could add it but I don't think this would make a difference.

> maybe mention it in the contribution section in the docs.
Definitely we should. I'm not quite how to mention this in a legal way. The contributor license agreement of the CNCF seems like a massive overkill to me...